### DISCUSSION TITLE
What's the difference between openmosaic vs pyart.map.map_to_grid?

### DISCUSSION THREAD

**Discussion by ahuang11** [jthielen/OpenMosaic#7] [General]
I only heard about this package from a colleague at AMS so I was wondering is there a difference between this package and http://arm-doe.github.io/pyart/API/generated/pyart.map.map_to_grid.html?

Is OpenMosaic more optimized (I see some cython)?

http://arm-doe.github.io/pyart/API/generated/pyart.map.map_to_grid.html

**Comment by jthielen** [comment]
For full information at this point, I encourage you to check out my recorded talk on YouTube: https://www.youtube.com/watch?v=OQlnL_h8PYM

In summary, OpenMosaic refactors and extends Py-ART's `map_to_grid` to support 4D (space and time) weighting using GridRad's algorithm (which is needed for wide-ranging mosaics with scans at varying times) as well as only gridding single radars to subsets of the target grid. These individual "subgrid tasks" are meant to be handled with dask futures so that they can be readily parallelized in a highly-scalable way (whereas Py-ART's current routines only loop over individual radars with Python loops). Right now, as shown in the example scripts, this is set up to be performed on the script level, but I plan to make this easier to use in the future through direct integration into the Gridder class.

The low-level cython in OpenMosaic is no more or less optimized than that of Py-ART (since that's what I used as my starting point). Where OpenMosaic is optimized is its ability to leverage dask.
